strings/strings.go

```diff
@@ -133,6 +134,70 @@ func CaesarEncrypt(input string, shift int) string {
 	return string(shifted)
 }
 
+// Encode takes a string and returns its Run-Length Encoded representation
+func RunLengthEncode(input string) string {
+	if len(input) == 0 {
+		return ""
+	}
+
+	var encoded strings.Builder
+
+	currentChar := input[0]
+	count := 1
+
+	for i := 1; i < len(input); i++ {
+		if input[i] == currentChar {
+			count++
+		} else {
+			encoded.WriteByte(currentChar)
+			encoded.WriteString(strconv.Itoa(count))
+			currentChar = input[i]
+			count = 1
+		}
+	}
+
+	// Write the last character and its count
+	encoded.WriteByte(input[len(input)-1])
+	encoded.WriteString(strconv.Itoa(count))
+
+	// Return the original string if encoding doesn't save space
+	if encoded.Len() >= len(input) {
+		return input
+	}
+
+	return encoded.String()
+}
+
+// Decode takes a Run-Length Encoded string and returns the decoded original string
+func RunLengthDecode(encoded string) string {
+	if len(encoded) == 0 {
+		return ""
+	}
+
+	var decoded strings.Builder
+	length := len(encoded)
+	for i := 0; i < length; {
+		char := encoded[i]
+		j := i + 1
+
+		// Extract the number after the character
+		for j < length && encoded[j] >= '0' && encoded[j] <= '9' {
+			j++
+		}
+
+		// Parse the count
+		count, _ := strconv.Atoi(encoded[i+1 : j])
+
+		// Write the character `count` times
+		decoded.WriteString(strings.Repeat(string(char), count))
+
+		// Move to the next character group
+		i = j
+	}
+
+	return decoded.String()
+}
+
```

Comment:
this fucntion seems a bit ambiguous to me. either we should have validation in place or at least add detailed comment on the function with instructions on what to send as argument.. For example, what does this mean?
```
RunLengthDecode("1122")
```
Does this mean eleven times twenty-two or hundred and twelve times two?

May be I am not fully understanding this one so please correct me if I am wrong.

Reply:
RLE does not encode numbers. There is a possible implementation of RLE that could encode numbers, but in most cases it does not lead to an efficient result